src/state/slices/time_slice.ts

```diff
@@ -76,9 +92,15 @@ export const createTimeSlice: StateCreator<TimeSlice & DatasetSlice, [], [], Tim
     }
     set({ pendingFrame: frame });
     await get()
-      .loadFrameCallback(frame)
-      .then(() => {
-        set({ currentFrame: frame });
+      .frameLoadCallback(frame)
+      .then((result) => {
+        if (result !== null) {
+          set({ currentFrame: result.frame });
+        } else if (get().pendingFrame === frame) {
+          // Reset pendingFrame if it hasn't changed. (e.g. no other calls to
+          // setFrame were made while this one was loading)
+          set({ pendingFrame: get().currentFrame });
+        }
```

Comment:
Note also, I didn't set `frameLoadResult` in this same step. I ended up having all changes to it go through the callback set in `Viewer.tsx`, because there are situations where `ColorizeCanvas` calls `setFrame` on itself (like when Datasets are being changed). Because those `setFrame` calls aren't being called from here in the TimeSlice, the slice would have no way to know the result of the most recent frame load request.

This is why there's still a callback in `ColorizeCanvas/IRenderCanvas` for when frames are loaded-- I had a discussion with Cameron and we agree that this pattern makes sense because it's like an event handler/event listener.